CorsixTH/Src/th_movie.cpp

```diff
@@ -522,6 +522,17 @@ void movie_player::play_audio(int requested_audio_channel) {
 
 void movie_player::stop() { aborting = true; }
 
+void movie_player::togglePause() {
+  bool wasPaused = paused.exchange(!paused);
```

Comment:
This is so close to thread safe, if c++ had an atomic not, or if fetch_xor worked on bools I would have done it. As is I'd need a mutex or a compare_exchange loop and it didn't seem worth it since we only call from the event thread anyway.